pkg/instrument/extension.h

```diff
@@ -0,0 +1,39 @@
+#include <sqlite3.h>
+#include <stdint.h>
+
+typedef int sqlite3_error;
+
+typedef struct sqlite3_metrics_s {
+    uint64_t pages_cache_write;
+    uint64_t pages_cache_hit;
+    uint64_t pages_cache_miss;
+    uint64_t pages_cache_spill;
+
+    uint64_t read_txn_time_ns;
+    uint64_t write_txn_time_ns;
+} sqlite3_metrics_t;
+
+// sqlite3_instrument register an auto extension that 
+// instruments all new connections to collect sqlite
+// performance metrics. Existing connections will not be 
+// affected.
+// This call uses the trace hook to collect data. Replacing
+// the hook will remove instrumentation for the connection. 
+// In this case, it is possible to call [sqlite3_collect_metrics]
+// to keep the instrumentation active in a custom trace hook.
+sqlite3_error sqlite3_instrument();
+
+// sqlite3_deinstrument removes instrumentation connection
+// and free the associated context.
+// It will not restore previous commit or trace hook.
+// If an error occurs, the conneciton is left unchanged
+// and the instrumentation context is not freed.
+void sqlite3_deinstrument();
```

Comment:
I think to match the previous function, we should mention removing instrumentation from 'existing connections, freeing their associated context.' I'm not sure what the associated context is, I think it'd be better to clarify or drop this.

In the second sentence, the 'it' s not immediately clear, perhaps 'Calling this function'?

In the 'If an error occurs' paragraph, would 'existing connections are left unchanged work'? (Also there's a typo on connection :P)

Reply:
I think this is coming again from a previous version. I simplified with:

```
// sqlite3_deinstrument stops instrumenting new connections.
// Existing connections are left unchanged.
```